File: src/shared/Infrastructure/Security/Permissions.cs
namespace Infrastructure.Security;

/// <summary>
/// Permissions for fine-grained access control
/// </summary>
public static class Permissions
{
    // User Management
    public const string ReadUsers = "users:read";
    public const string WriteUsers = "users:write";
    public const string DeleteUsers = "users:delete";
    public const string ManageUserRoles = "users:manage_roles";

    // Skill Management
    public const string ReadSkills = "skills:read";
    public const string WriteSkills = "skills:write";
    public const string DeleteSkills = "skills:delete";
    public const string ManageCategories = "skills:manage_categories";

    // Matching
    public const string AccessMatching = "matching:access";
    public const string ViewAllMatches = "matching:view_all";

    // Appointments
    public const string ReadAppointments = "appointments:read";
    public const string WriteAppointments = "appointments:write";
    public const string DeleteAppointments = "appointments:delete";

    // Video Calls
    public const string AccessVideoCalls = "videocalls:access";
    public const string ManageVideoCalls = "videocalls:manage";

    // System
    public const string AccessAdminPanel = "system:admin_panel";
    public const string ViewSystemLogs = "system:logs";
    public const string ManageSystem = "system:manage";
}

File: src/client/src/core/config/permissions.ts
export const Permissions = {
  // User Management
  READ_USERS: 'users:read',
  WRITE_USERS: 'users:write',
  DELETE_USERS: 'users:delete',
  MANAGE_USER_ROLES: 'users:manage_roles',

  // Skill Management
  READ_SKILLS: 'skills:read',
  WRITE_SKILLS: 'skills:write',
  DELETE_SKILLS: 'skills:delete',
  MANAGE_CATEGORIES: 'skills:manage_categories',

  // Matching
  ACCESS_MATCHING: 'matching:access',
  VIEW_ALL_MATCHES: 'matching:view_all',

  // Appointments
  READ_APPOINTMENTS: 'appointments:read',
  WRITE_APPOINTMENTS: 'appointments:write',
  DELETE_APPOINTMENTS: 'appointments:delete',

  // Video Calls
  ACCESS_VIDEO_CALLS: 'videocalls:access',
  MANAGE_VIDEO_CALLS: 'videocalls:manage',

  // System
  ACCESS_ADMIN_PANEL: 'system:admin_panel',
  VIEW_SYSTEM_LOGS: 'system:logs',
  MANAGE_SYSTEM: 'system:manage',
} as const;

export type Permission = (typeof Permissions)[keyof typeof Permissions];
